include/rocksdb/comparator.h

```diff
@@ -89,4 +89,41 @@ extern const Comparator* BytewiseComparator();
 // ordering.
 extern const Comparator* ReverseBytewiseComparator();
 
+// The below comparators are compatible with BytewiseComparator(): they have
+// the Name() and do comparison exactly the same way. They differ only
+// in FindShortestSeparator() and FindShortSuccessor(), which have an effect
+// on sst file indexes. These comparators can improve iterator seek performance
+// in some situations, but may increase index size.
+
+// Same as BytewiseComparator, except that FindShortSuccessor() does nothing.
+// This can improve iterator seek performance when seeking to a key higher
+// than the highest key in some of the sst files.
+// May make sst index slightly bigger.
+//
+// The sst index contains an upper bound of the highest key in the file.
+// This upper bound is obtained using FindShortSuccessor(). If it's
+// overestimated by a lot, an iterator seek above the highest key in the file
+// may unnecessarily read the last data block. This is particularly noticeable
+// when block cache is disabled (ReadOptions::fill_cache = false) and direct IO
+// is enabled (use_direct_reads = true).
```

Comment:
I'm curious how this can happen. My understanding for levels above L0, we compare with smallest and largest key of files stored in manifest to decide whether to read from a file. Am I missing something?

Reply:
It's in L0.